### PR TITLE
Ensure reference is resolved when passing options to from_param

### PR DESCRIPTION
**Pull request by philippjfr**
Fixes https://github.com/holoviz/panel/issues/6743

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def run(self):
 
 install_requires = [
     'bokeh >=3.4.0,<3.5.0',
-    'param >=2.0.0,<3.0',
+    'param >=2.1.0,<3.0',
```
Comment (Hoxbro) [inline setup.py:116]: Are you sure that we want to bump versions in a micro release?

Comment (philippjfr) [reply setup.py:116]: I would argue bumping versions to resolve bugs is appropriate in a patch release, but I'm open to alternative opinions.

Comment (Hoxbro) [reply setup.py:116]: My feeling is that if it were a bug in a 3rd party library like Pandas, we would not bump the version. We would likely do a version check, and accept the shortcomings if you are not using a version supporting the bug.

Comment (philippjfr) [reply setup.py:116]: Depends on the severity of the bug I suppose. If it was just this bug I'd definitely postpone the bump but there's a variety of other bugs that are more far reaching. Really should have bumped in 1.4.0 already.

**Pull request merged**